core-bundle/src/Controller/SitemapController.php

```diff
@@ -110,4 +122,55 @@ private function callLegacyHook(PageModel $rootPage, array $pages): array
 
         return $pages;
     }
+
+    private function getPageAndArticleUrls(int $pid): array
```

Comment:
Shouldn't we use individual `SitemapEvent` listeners for articles and pages?

Reply:
This would cause multiple issues, e.g. the pages would no longer be in the legacy hook.